test/named-parameter-order-test.js

```diff
@@ -0,0 +1,36 @@
+var shuffle = require('shuffle-array');
+var test = require('tap').test;
+var docuri = require('..');
+
+var simulateWrongOrderObjectKeys = function(cb){
+  var orig = Object.keys;
+  Object.keys = function(obj){
+    var keys = orig(obj);
+    var real = keys.toString();
+    while (real === keys.toString())
+      shuffle(keys);
```

Comment:
That would not be enough for the test.
It would allow the implementation to rely on that arbitrary order, even if it's not the insertion order.
The point is the order can change _every time_ you call `Object.keys`.

To prevent tests accidentally passing, I have the `real === keys.toString()` check.

The dependency is only a dev dependency, so has no influence on consumers.